numpy/f2py/cfuncs.py

```diff
@@ -320,10 +320,10 @@
     'pyobj_from_complex_float1'] = '#define pyobj_from_complex_float1(v) (PyComplex_FromDoubles(v.r,v.i))'
 needs['pyobj_from_string1'] = ['string']
 cppmacros[
-    'pyobj_from_string1'] = '#define pyobj_from_string1(v) (PyString_FromString((char *)v))'
```

Comment:
This one worries me a bit, since the rest of this file uses `PyString_Check` a lot, which might no longer be satisfied.

Reply:
I was a also a bit uncertain about that. I think it is intended for returning strings to Python. Not sure how to check it without looking at some generated wrappers. @pearu Thoughts?

Reply:
I suppose it could cause problems if the user has worked around returned byte strings.

Reply:
I've looked at this some more and think it the right thing to do here. It is a change, but I think likely harmless, and if it actually turns up in applications I think the fix would probably improve the code.

Reply:
Thanks, should we add a small release note in that case (also helps me with understanding what the change is)?

Reply:
Scanning the files for `pyobj_from_` shows that the macro `pyobj_from_string1` is likely never used in wrapper generation, only `pyobj_from_string1size` and `pyobj_from_p_string1` are used.

Reply:
Added a small release note.